lib/ontologies_api_client/models/class.rb

```diff
@@ -61,10 +65,30 @@ def self.find(id, ontology, params = {})
 
         def self.search(*args)
           query = args.shift
+
           params = args.shift || {}
+
           params[:q] = query
+          params[:is_collection] = true
+
           raise ArgumentError, "You must provide a search query: Class.search(query: 'melanoma')" if query.nil? || !query.is_a?(String)
-          HTTP.post("/search", params)
+
+          if params[:federate] || params[:portals]
+            search_result = federated_get(params) do |url|
+              "#{url}/search"
+            end
+            merged_collections = {results: [], errors: []}
+            search_result.each do |result|
+              if result.collection
+                merged_collections[:results].concat(result.collection)
+              elsif result.errors
+                merged_collections[:errors] << result.errors
```

Comment:
need to check if we handle the errors in the same way in this https://github.com/ontoportal-lirmm/ontologies_api_ruby_client/pull/17